src/xai_grok/schemas/tool_choice.py

```diff
@@ -0,0 +1,10 @@
+import typing
+
+import pydantic
+
+from xai_grok.schemas.function_tool_choice import FunctionToolChoice
+
+
+class ToolChoice(pydantic.BaseModel):
+    type: typing.Literal["function"]
```

Comment:
The `function` field is marked as optional with a default of `None`, but the `type` field is set to a literal `"function"`. This creates an inconsistency where the type indicates a function tool is being used, but the actual function data could be missing.